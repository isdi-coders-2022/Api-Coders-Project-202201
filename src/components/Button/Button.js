import styled from "styled-components";

const StarButton = styled.button`
  background-color: transparent;
  color: white;
  height: 50px;
  width: 100px;
  border: none;
  font-size: 30px;
`;

const Button = ({ text, actionOnClick }) => {
  return <StarButton onClick={actionOnClick}>{text}</StarButton>;
};

export default Button;
