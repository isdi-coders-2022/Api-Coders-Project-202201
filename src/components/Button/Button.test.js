import { render, screen } from "@testing-library/react";
import Button from "./Button";
import reactTestRenderer from "react-test-renderer";

describe("Given a Button component", () => {
  describe("When it's rendered", () => {
    test("Then it should display the text passed", () => {
      const expectedText = "Let's see";
      render(<Button text={expectedText} />);

      const renderedButton = screen.getByText(expectedText);

      expect(renderedButton).toBeInTheDocument();
    });
  });
  describe("When it get's passed a function", () => {
    test("Then it should call that function", () => {
      const defaultText = "Let's see";
      const calledFunction = jest.fn();

      render(<Button text={defaultText} action={calledFunction} />);

      const renderedButton = screen.getByText(defaultText);

      renderedButton.click();

      expect(calledFunction).toHaveBeenCalled();
    });
  });
});

it("renders consistently , equal to the snapshot", () => {
  const theButton = reactTestRenderer.create(<Button />);

  expect(theButton.toJSON()).toMatchSnapshot();
});
