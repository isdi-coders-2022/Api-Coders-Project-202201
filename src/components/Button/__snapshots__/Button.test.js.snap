// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Button component When it's rendered' Then it should always match this snapshot 1`] = `
<button
  className="sc-bdvvtL dZIbqa"
/>
`;
